torch/_prims/utils.py

```diff
@@ -180,18 +183,143 @@ def compare_tensor_meta(a: TensorLikeType, b: TensorLikeType):
+            str(b.device) == "cuda:0" or str(b.device) == "cuda"
+        ):
+            pass
+        else:
+            msg = "Devices {0} and {1} are not equal!".format(a.device, b.device)
+            raise AssertionError(msg)
+
+    same_strides, idx = check_significant_strides(a, b)
+    if not same_strides:
+        msg = "Stride mismatch! Strides are {0} and {1} (mismatched at {2})!".format(
+            a.stride(), b.stride(), idx
+        )
+        raise RuntimeError(msg)
+
 
+def check_significant_strides(
+    a: TensorLikeType, b: TensorLikeType
+) -> Tuple[bool, Optional[int]]:
+    # NOTE: only on CUDA because CPU elementwise strides are incorrect in PyTorch
+    # See https://github.com/pytorch/pytorch/issues/77553
+    # Only compares strides that are "meaningful" -- strides for dimensions with length > 1
+    # and for tensors with more than one element
+    if (a.device.type == "cuda" or b.device.type == "cuda") and a.numel() > 0:
+        for idx in range(a.ndim):
+            if a.stride()[idx] != b.stride()[idx] and a.shape[idx] > 1:
+                return False, idx
 
-def compare_significant_strides(a: TensorLikeType, b: TensorLikeType):
-    assert a.ndim == b.ndim
+    return True, None
 
-    for idx in range(a.ndim):
-        assert a.shape[idx] == b.shape[idx]
-        if a.shape[idx] == 0 or a.shape[idx] == 1:
+
+def is_contiguous(a: TensorLikeType) -> bool:
+    """
+    Tests whether a tensor is contiguous or not.
+
+    Tensors are contiguous when they have no elements,
+    or when they have "nested" strides.
+    """
+    if a.numel() == 0:
+        return True
+
+    expected_stride = 1
+    for x, y in reversed(tuple(zip(a.shape, a.stride()))):
+        # Skips checking strides when a dimension has length 1
+        if x == 1:
             continue
-        assert a.stride()[idx] == b.stride()[idx]
+
+        if y != expected_stride:
+            return False
+        expected_stride = expected_stride * x
+
+    return True
+
+
+# NOTE: Based on the implementation in TensorIterator.cpp, but note that
+# the note [Computing output strides] is incorrect, because it
+# says that strides will be preserved even if they are not
+# "non overlapping and dense", but this is incorrect. The
+# output of elementwise operations are always given
+# non overlapping and dense strides.
+# This is also INCORRECT because it does not model TensorIterator's
+# short-circuit, which can cause different strides.
+def compute_elementwise_output_strides(*tensors) -> Tuple[int, ...]:
+    """
+    Computes the output strides for elementwise operations.
+    """
+
+    if len(tensors) == 0:
+        msg = "Can't compute elementwise output strides for zero tensors!"
+        raise ValueError(msg)
+
+    check_same_shape(*tensors, allow_cpu_scalar_tensors=True)
+
+    # Filters the tensors to actual tensors
+    all_tensors = all(isinstance(a, TensorLike) for a in tensors)
```

Comment:
`all_tensors` is not used.

Reply:
doh! you're right -- thanks @jjsjann123! I'll get the cleaned up